indra/statements.py

```diff
@@ -36,8 +73,7 @@ def matches(self, other):
 
     def matches_key(self):
         key = (self.name,
-               set(self.mods),
-               set(self.mod_sites),
+               set([m.matches_key() for m in self.mods]),
```

Comment:
Maybe add a note that this may break if you get an agent with two phosphorylations at serine with unknown sites (as we see for MEK, for example).